controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -816,7 +815,7 @@ public void close() throws Exception {
 
     public String retrieveDelegationToken() {
         if (authEnabled) {
```

Comment:
Here, and above, and below, the 'if (authEnabled)' block is repeated. I think it would be better to move these methods out behind an interface, and have two implementations. One for auth enabled and one without auth. That way this conditional logic does not need to be repeated.

Reply:
The `authEnabled` is used only at one place. Do you think converting to an interface still makes sense?

Reply:
It appears in several places. See StreamTransactionMetadataTasks:685, ControllerServiceImpl:371, 389, 397 
There is no harm in having multiple small classes. My above comments on ControllerServiceImpl would not have come up if the auth code had it's own class as then it would be more clear what it was doing and there would be a clear place for documentation.

Reply:
Refactored the auth handling code to a helper class.